### DISCUSSION TITLE
random_nauts not working

### DISCUSSION THREAD

**Discussion by peyre** [raceintospace/raceintospace#647] [Q&A] [answered]
I've been testing some of the other settings in the config file.  It looks like **random_nauts** doesn't actually work.  I turned it on, and the first 7 cosmonauts are still all CA2 or above.  Does anyone know if this is an option that was never implemented, or am I missing something?

![RandomNauts](https://user-images.githubusercontent.com/3118014/168483967-33ee0e46-92de-4063-8c13-e817e7977b95.png)

**Comment by peyre** [comment]
For that matter, **audio** and **nofail** don't seem to work either.

**Comment by peyre** [comment]
Also, **compt_nauts 1** is supposed to make all your spacepeople get along, but it doesn't seem to actually do it.

![compt_nauts](https://user-images.githubusercontent.com/3118014/168484423-832f0ed0-a902-40fd-93b6-50b1a7ee2327.png)

**Comment by peyre** [comment]
Also, **boosterSafety** seems to have a problem. **boosterSafety 1** is supposed to use "Minimum Safety" - I assume the lower of rocket and booster, and **boosterSafety 2** is supposed to make boosters the average of rocket+booster, like in BARIS.  Yet neither setting seems to work - they both still use statistical safety.  Even if it's set to Classic mode it still uses statistical.  In this game my R-7 is at 74% and boosters at 23%. No matter the setting I use, the boosted rocket always comes out at 17%, the statistical result.

![boosterSafety](https://user-images.githubusercontent.com/3118014/168485331-a5ab7d88-e8fd-4d26-9a96-500410687ce1.png)

**Comment by peyre** [comment]
Hang on a minute.  I think there's something wrong with the changes I made to options.cpp yesterday.  I downloaded the old options.cpp, recompiled the game, and (at least some of) the options are working now.

Maybe I should revert my commit from yesterday!

**Reply by rnyoakum** [reply]
Yeah, commit 1f48cbb7c894ddcbb1b1767ccaf1de4c42ecdc01 is the problem. In trying to list the default values, you changed the _variable name_ in the config file. Consequently, the options.cpp code won't recognize, say, `compt_nauts 3` because it considers `compt_nauts 10` to be the name of the option.

**Comment by rnyoakum** [comment, marked as answer]
> It looks like random_nauts doesn't actually work. I turned it on, and the first 7 cosmonauts are still all CA2 or above. Does anyone know if this is an option that was never implemented, or am I missing something?

Randomized astro/cosmonauts are not currently set to trigger for human players. It does trigger for the AI, however. Most of the code is available, so it would be simple to enable it. However, the current randomization system is _very_ random and likely to produce more skilled recruits than the historical roster. Also, I was wondering if it should be implemented as a game option (historical/custom/random) rather than as a config file option?

> Also, compt_nauts 1 is supposed to make all your spacepeople get along, but it doesn't seem to actually do it.

The  **compt_nauts** option only affects new recruits, so any crew who were already recruited will not be affected by a change in the option. It's a frustrating option, and I have some ideas about how crew compatibility should be changed. Did you experience this problem with new recruits, or only existing crew?

> Also, boosterSafety seems to have a problem.

I have no immediate answer to this. I'll have to look into it. If I have the same problem, I'll open an issue.
EDIT: I checked this, and I had no problem on my build. As you suspected, https://github.com/raceintospace/raceintospace/commit/1f48cbb7c894ddcbb1b1767ccaf1de4c42ecdc01 effectively disabled the option.

**Comment by peyre** [comment]
Thank you Ryan!  Wow, now you explain it it makes sense.  I'll have to fix it, thanks.

Not sure about compt_nauts - I have to test that again, and make sure I recruit while that option is enabled.

boosterSafety looks to be working properly, once I replaced my messed-up version of options.cpp, so you can ignore this one.

I wouldn't be inclined to make random_nauts into a main Preferences option; I wouldn't expect it to be used much.  I'm open to the idea though if you think it would be a good idea.  If we leave it as it, we could just update the comment to explain that it only randomizes the AI's candidates.

**Comment by peyre** [comment]
I just checked compt_nauts, and sure enough it works on new recruits.  Thanks Ryan.
